API-Samples/Java/app/src/main/java/eventprocessor/SimpleCheckpointStore.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package eventprocessor;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.eventhubs.CheckpointStore;
+import com.azure.messaging.eventhubs.models.Checkpoint;
+import com.azure.messaging.eventhubs.models.PartitionOwnership;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.OWNER_ID_KEY;
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.PARTITION_ID_KEY;
+import static com.azure.messaging.eventhubs.implementation.ClientConstants.SEQUENCE_NUMBER_KEY;
+
+/**
+ * Source: <a href="https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/SampleCheckpointStore.java">...</a>
+ * A simple in-memory implementation of a {@link CheckpointStore}. This implementation keeps track of partition
+ * ownership details including checkpointing information in-memory. Using this implementation will only facilitate
+ * checkpointing and load balancing of Event Processors running within this process.
+ */
+public class SimpleCheckpointStore implements CheckpointStore {
+
+    private static final String OWNERSHIP = "ownership";
+    private static final String SEPARATOR = "/";
+    private static final String CHECKPOINT = "checkpoint";
+    private final Map<String, PartitionOwnership> partitionOwnershipMap = new ConcurrentHashMap<>();
+    private final Map<String, Checkpoint> checkpointsMap = new ConcurrentHashMap<>();
+    private static final ClientLogger LOGGER = new ClientLogger(SimpleCheckpointStore.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Flux<PartitionOwnership> listOwnership(String fullyQualifiedNamespace, String eventHubName,
+                                                  String consumerGroup) {
+        LOGGER.info("Listing partition ownership");
+
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup, OWNERSHIP);
+        return Flux.fromIterable(partitionOwnershipMap.keySet())
+                .filter(key -> key.startsWith(prefix))
+                .map(key -> partitionOwnershipMap.get(key));
+    }
+
+    private String prefixBuilder(String fullyQualifiedNamespace, String eventHubName, String consumerGroup,
+                                 String type) {
+        return new StringBuilder()

```

Comment:
Optional: Can be done with String format

Reply:
sample code from Azure Git hub -> don't want to change that.